core/src/main/java/org/bitcoinj/utils/AppDataDirectory.java

```diff
@@ -45,19 +54,36 @@ public static Path get(String appName) {
         return applicationDataDirectory;
     }
 
-    private static Path getPath(String appName) {
+    /**
+     * Return the Path to the application data directory without making
+     * sure it exists or creating it. (No disk I/O)
+     *
+     * @param appName The name of the current application
+     * @return Path to the application data directory
+     */
+    public static Path getPath(String appName) {
         final Path applicationDataDirectory;
 
         if (Utils.isWindows()) {
-            applicationDataDirectory = Path.of(System.getenv("APPDATA"), appName);
+            applicationDataDirectory = pathOf(System.getenv("APPDATA"), appName.toLowerCase());
```

Comment:
Are these `toLowerCase()` backward compatible? Or could it perhaps happen that wallettemplate doesn't find its wallets created by an earlier version any more on case-sensitive filesystems?

Reply:
> Are these toLowerCase() backward compatible?

No, but I'm pretty sure the AppDataDirectory stuff hasn't been released yet. And WalletTemplate isn't (yet) something we have to worry about that kind of backward-compatibility for (IMO.)

Reply:
Ok, you're right, I forgot. Still I think the `.toLowerCase()` change should be put into a separate commit because it seems unrelated to the refactoring described in this PR.

Reply:
I'm happy to put it in a separate commit, but I do consider it's absence a bug and was shocked that I had missed it. The whole idea was to replicate the behavior of Bitcoin Core.

Reply:
Even more so it could be a separate bugfix that can be merged much quicker than the proposed refactoring.

Reply:
OK, will do. The bug doesn't exist in the App Dir code in `core` because that code hardcodes `bitcoin` or `Bitcoin` as appropriate, but it does exist in the `wallettemplate` `AppDataDirectory` class...

Reply:
Fixed in PR #1882